packages/agent/src/oidc-v2.ts

```diff
@@ -0,0 +1,689 @@
+import { Convert, RequireOnly } from '@web5/common';
+import {
+  Ed25519,
+  EdDsaAlgorithm,
+  JoseHeaderParams,
+  Jwk,
+  Sha256,
+  utils,
+  X25519,
+} from '@web5/crypto';
+
+import { appendPathToUrl } from './utils.js';
+import { Hkdf } from './prototyping/crypto/primitives/hkdf.js';
+import { xchacha20poly1305 } from '@noble/ciphers/chacha';
+import type { ConnectPermissionRequests } from './connect-v2.js';
+import { DidDht, DidDocument, type BearerDid } from '@web5/dids';
+
+/**
+ * Sent to an OIDC server to authorize a client. Allows clients
+ * to securely send authorization request parameters directly to
+ * the server via POST. This avoids exposing sensitive data in URLs
+ * and ensures the server validates the request before user interaction.
+ *
+ * @see {@link https://www.rfc-editor.org/rfc/rfc9126.html | OAuth 2.0 Pushed Authorization Requests}
+ */
+export type PushedAuthRequest = {
+  /** The JWT which contains the {@link HybridAuthRequest} */
+  request: string;
+};
+
+/**
+ * Sent back by OIDC server in response to {@link PushedAuthRequest}
+ * The server generates a TTL and a unique request_uri. The request_uri can be shared
+ * with the Provider using a link or a QR code along with additional params
+ * to access the url and decrypt the payload.
+ */
+export type PushedAuthResponse = {
+  request_uri: string;
+  expires_in: number;
+};
+
+/**
+ * An auth request that is compatible with both Web5 Connect and (hopefully, WIP) OIDC SIOPv2
+ * The contents of this are inserted into a JWT inside of the {@link PushedAuthRequest}.
+ */
+export type HybridAuthRequest = SIOPv2AuthRequest & Web5ConnectRequest;
```

Comment:
Consider alternative: reads as "web5 connect request is a SIOPv2 auth request + some other custom things".

```suggestion
export type Web5ConnectRequest = SIOPv2AuthRequest & {
  /** PermissionGrants that are to be sent to the provider */
  permission_requests: ConnectPermissionRequests;
};
```

Reply:
done